integrationtest/orm/src/test/java/org/hibernate/search/v6poc/integrationtest/orm/OrmUnusedPropertiesIT.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.v6poc.integrationtest.orm;
+
+import java.util.function.Consumer;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.Metadata;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.SessionFactoryBuilder;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.search.v6poc.entity.orm.cfg.SearchOrmSettings;
+import org.hibernate.search.v6poc.entity.pojo.mapping.definition.annotation.DocumentId;
+import org.hibernate.search.v6poc.entity.pojo.mapping.definition.annotation.Field;
+import org.hibernate.search.v6poc.entity.pojo.mapping.definition.annotation.Indexed;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.rule.BackendMock;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.rule.ExpectedLog4jLog;
+import org.hibernate.search.v6poc.util.impl.integrationtest.common.stub.backend.index.impl.StubBackendFactory;
+import org.hibernate.service.ServiceRegistry;
+
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class OrmUnusedPropertiesIT {
+
+	private static final String PREFIX = SearchOrmSettings.PREFIX;
+
+	@Rule
+	public BackendMock backendMock = new BackendMock( "stubBackend" );
+
+	@Rule
+	public ExpectedLog4jLog log = ExpectedLog4jLog.create();
+
+	private SessionFactory sessionFactory;
+
+	@After
+	public void cleanup() {
+		if ( sessionFactory != null ) {
+			sessionFactory.close();
+		}
+	}
+
+	@Test
+	public void checkDisabled_unusedProperty() {
+		String unusedPropertyKey = "hibernate.search.index.default.foo";
+		log.expectMessageMissing(
+				"Some properties in the Hibernate Search configuration were not used"
+		);
+		log.expectMessage( "Configuration property tracking is disabled" );
+		setup( builder -> {
+			builder.applySetting( SearchOrmSettings.ENABLE_CONFIGURATION_PROPERTY_TRACKING, false );
+			builder.applySetting( unusedPropertyKey, "bar" );
+		} );
+	}
+
+	@Test
+	public void checkEnabledByDefault_unusedProperty() {
+		String unusedPropertyKey = "hibernate.search.index.default.foo";
+		log.expectMessage(
+				"Some properties in the Hibernate Search configuration were not used",
+				"[" + unusedPropertyKey + "]"
+		);
+		log.expectMessageMissing( "Configuration property tracking is disabled" );
```

Comment:
Shouldn't we also test that a consumed property key is not logged?

Reply:
It's already covered by the current test, since there is a single message for all properties and we constrained it to contain `"[" + unusedPropertyKey + "]"`, which means the list of properties only includes `unusedPropertyKey`.

I added this to be extra sure:

```
		// Also check that used properties are not reported as unused
		log.expectMessageMissing(
				"not used",
				DEFAULT_BACKEND_PROPERTY_KEY
		);
```